### ISSUE TITLE
Dialogs should have proper a11y markup

### ISSUE THREAD

**Issue by adidahiya** [palantir/blueprint#4528]
#### Feature request

Dialogs should have `role="dialog"`, `aria-labeledby`, and optionally `aria-describedby` attributes for accessibility. At the very least there should be docs about how to configure this markup properly.

#### References

- https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/dialog_role

**Comment by adidahiya** [comment]
Also potentially useful to document: `data-first` and `data-last` attributes for dialog content

**Comment by jabsatz** [comment]
I would need this for a project I'm working on, since I'm using react-testing-library and would have to access dialogs by role. I'm preparing a solution and hopefully have a PR soon so it can be reviewed.
